packages/react-components/react-motion/src/factories/createPresenceComponent.ts

```diff
@@ -45,18 +45,24 @@ function shouldSkipAnimation(appear: boolean | undefined, isFirstMount: boolean,
   return !appear && isFirstMount && visible;
 }
 
-export function createPresenceComponent(value: PresenceMotion | PresenceMotionFn) {
-  const Presence: React.FC<PresenceComponentProps> = props => {
+export function createPresenceComponent<MotionParams extends Record<string, MotionParam> = {}>(
+  value: PresenceMotion | PresenceMotionFn<MotionParams>,
+) {
+  const Presence: React.FC<PresenceComponentProps & MotionParams> = props => {
     const itemContext = React.useContext(PresenceGroupChildContext);
-    const { appear, children, imperativeRef, onMotionFinish, visible, unmountOnExit } = { ...itemContext, ...props };
+    const { appear, children, imperativeRef, onExit, onMotionFinish, visible, unmountOnExit, ..._rest } = {
```

Comment:
```suggestion
    const merged = { ...itemContext, ...props };
    const { appear, children, imperativeRef, onExit, onMotionFinish, visible, unmountOnExit, ..._rest } = merged;
    const params = _rest as Exclude<typeof merged, PresenceComponentProps | typeof itemContext>;
```
This is a bit safer because we're no longer casting through `unknown`.